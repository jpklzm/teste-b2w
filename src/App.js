import React, {Component} from 'react';
import styles from './App.module.css';
import Planet from './components/planet/planet.js';

    class App extends Component {
      constructor(props){
        super()

        this.onClick = this.onClick.bind(this);
        this.getPlanet = this.getPlanet.bind(this);
        this.getRandomColor = this.getRandomColor.bind(this);
      }

      getPlanet(requestUrl){
        fetch(requestUrl)
        .then(res => res.json())
        .then((data) => {
          this.setState({ planet: data, color: this.getRandomColor() })
        })
        .catch(console.log)
      }


      getRandomColor(){
        const getColor = Math.floor(Math.random() * 4 + 1)
        switch(getColor) {
          case 1:
            return styles.red;
          case 2:
            return styles.blue;
          case 3:
            return styles.green;
          case 4:
            return styles.purple;
          default:
            return styles.purple;
        }
      }

      componentWillMount() {
        const requestUrl = 'https://swapi.co/api/planets/' + Math.floor(Math.random() * 61 + 1)   +  '/';
        this.getPlanet(requestUrl);
        this.setState({ color: this.getRandomColor() })
      }

      onClick(e) {
        const requestUrl = 'https://swapi.co/api/planets/' + Math.floor(Math.random() * 61 + 1)   +  '/';
        this.getPlanet(requestUrl);
      }


      render () {
        return (
          <div className={styles.content}>
            <center>
              <img src="https://www.freepnglogos.com/uploads/star-wars-logo-design-14.png" />
              <div>{this.state.planet ? <Planet planet={this.state.planet} color={this.state.color} /> : null}</div>
              <button onClick={this.onClick} className={this.state.color}>Next</button>
            </center>
          </div>
        );
      }
    }

    export default App;