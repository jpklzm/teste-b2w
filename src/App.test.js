import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});


it("renders correctly", () => {
  const wrapper = shallow(
    <App />
  );

  expect(wrapper).toMatchSnapshot();
});

it("check lifecycle", () => {
  const wrapper = mount(
    <App />
  );

  expect(wrapper).toMatchSnapshot();
});

it("renders childrens correctly", () => {
  const wrapper = render(
    <App />
  );

  expect(wrapper).toMatchSnapshot();
});