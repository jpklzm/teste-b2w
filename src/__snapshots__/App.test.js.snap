// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check lifecycle 1`] = `
<div
  className="content"
>
  <center>
    <img
      src="https://www.freepnglogos.com/uploads/star-wars-logo-design-14.png"
    />
    <div />
    <button
      className="purple"
      onClick={[Function]}
    >
      Next
    </button>
  </center>
</div>
`;

exports[`renders childrens correctly 1`] = `
<div
  class="content"
>
  <center>
    <img
      src="https://www.freepnglogos.com/uploads/star-wars-logo-design-14.png"
    />
    <div />
    <button
      class="green"
    >
      Next
    </button>
  </center>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="content"
>
  <center>
    <img
      src="https://www.freepnglogos.com/uploads/star-wars-logo-design-14.png"
    />
    <div />
    <button
      className="purple"
      onClick={[Function]}
    >
      Next
    </button>
  </center>
</div>
`;
